lib/go-tc/federation.go

```diff
@@ -184,7 +186,9 @@ type FederationUserPost struct {
 
 // Validate validates FederationUserPost
 func (f *FederationUserPost) Validate(tx *sql.Tx) error {
-	return nil
+	return validation.ValidateStruct(f,
+		validation.Field(&f.IDs, validation.NotNil),
```

Comment:
Does this also check for "not empty"?

Reply:
You know what, that doesn't matter. The best anyone could hope for passing `userIds` as `null` or `undefined` is that it would do nothing. Specific status codes in response to certain requests aren't reliably documented (intentionally) so this isn't even a breaking change.

Reply:
Perl did check if userIDs is defined and an array https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/lib/API/FederationUser.pm#L61

Reply:
Nice